earthpy/tests/test_spatial.py

```diff
@@ -108,6 +109,63 @@ def test_extent_to_json():
         es.extent_to_json([0, 1, 1, 0])
 
 
+def test_normalized_diff_shapes():
+    """If provided with arrays of different shapes,
+    normalized_diff returns ValueError."""
+
+    # Test data
+    b1_2d = np.array([[6, 7, 8, 9, 10], [16, 17, 18, 19, 20]])
+    b2_1d = np.array([[1, 2, 3, 4, 5, 14, 12, 13, 14, 17]])
+
+    # Check ValueError
+    with pytest.raises(ValueError, match=r"Both arrays should be of the same dimensions"):
+        es.normalized_diff(b1 = b1_2d, b2 = b2_1d)
+
+
+def test_normalized_diff_no_mask():
+    """If result of normalized_diff does not include nan values,
+    array is returned as unmasked."""
+
+    # Test data
+    b1_2d = np.array([[6, 7, 8, 9, 10], [16, 17, 18, 19, 20]])
+    b2_2d = np.array([[1, 2, 3, 4, 5], [14, 12, 13, 14, 17]])
+
+    n_diff = es.normalized_diff(b1 = b1_2d, b2 = b2_2d)
+
+    # Output array unmasked
+    assert ma.is_masked(n_diff) == False
+
+
+def test_normalized_diff_inf():
```

Comment:
Here i think you could use the same arrays just replace one value with -15

Reply:
Fixtures for these arrays have been implemented in testing of normalized_diff.